<launch>
    <arg name="car_name" default="bair_car" />
    <arg name="use_zed" default="false" />
    <arg name="use_AI" default="false" />
    
    <arg name="record" default="false" />
    <arg name="bagpath" default="/media/ubuntu/rosbags" />

    <group ns="$(arg car_name)">
        <node name="run_arduino" pkg="bair_car" type="arduino_node.py" output="screen">
        </node>


        <node name="run_AI" pkg="bair_car" type="get_ros_images_live2_z2_color_keras_TX1.py" output="screen" if="$(arg use_AI)">
        </node>

        <group ns="zed" if="$(arg use_zed)">        
            <include file="$(find zed_wrapper)/launch/zed_tf.launch" />

            <node name="zed_wrapper_node" pkg="zed_wrapper" type="zed_wrapper_node" args="" output="screen">

            <param name="resolution"            value="3" />
            <param name="quality"               value="0" />
            <param name="sensing_mode"          value="1" />
            <param name="frame_rate"            value="30" />
            <param name="odometry_DB"           value="" />
            <param name="openni_depth_mode"     value="0" />

            <param name="rgb_topic"             value="rgb/image_rect_color" />
            <param name="rgb_cam_info_topic"    value="rgb/camera_info" />
            <param name="rgb_frame_id"          value="/zed_tracked_frame" />

            <param name="left_topic"            value="left/image_rect_color" />
            <param name="left_cam_info_topic"   value="left/camera_info" />
            <param name="left_frame_id"         value="/zed_tracked_frame" />

            <param name="right_topic"           value="right/image_rect_color" />
            <param name="right_cam_info_topic"  value="right/camera_info" />
            <param name="right_frame_id"        value="/zed_tracked_frame" />

            <param name="depth_topic"           value="depth/image_rect_color" />
            <param name="depth_cam_info_topic"  value="depth/camera_info" />
            <param name="depth_frame_id"        value="/zed_tracked_frame" />

            <param name="point_cloud_topic"     value="point_cloud/cloud" />
            <param name="cloud_frame_id"        value="/zed_tracked_frame" />

            <param name="odometry_topic"                value="odom" />
            <param name="odometry_frame_id"             value="/zed_initial_frame" />
            <param name="odometry_transform_frame_id"   value="/zed_tracked_frame" />


            </node>
        </group>
        
    </group>
    
    <arg name="rosbag_args"
     value="/$(arg car_name)/cmd/motor /$(arg car_name)/cmd/state /$(arg car_name)/cmd/steer /$(arg car_name)/cmd/freeze /$(arg car_name)/encoder /$(arg car_name)/gps /$(arg car_name)/gyro /$(arg car_name)/gyro_heading /$(arg car_name)/acc /$(arg car_name)/motor /$(arg car_name)/sonar /$(arg car_name)/state /$(arg car_name)/steer /$(arg car_name)/GPS2_hour /$(arg car_name)/GPS2_min /$(arg car_name)/GPS2_sec /$(arg car_name)/GPS2_day /$(arg car_name)/GPS2_mon /$(arg car_name)/GPS2_yr /$(arg car_name)/GPS2_fix /$(arg car_name)/GPS2_qual /$(arg car_name)/GPS2_lat /$(arg car_name)/GPS2_long /$(arg car_name)/GPS2_speed /$(arg car_name)/GPS2_angle /$(arg car_name)/GPS2_alt /$(arg car_name)/GPS2_sat /$(arg car_name)/camera_heading /$(arg car_name)/zed/depth/camera_info /$(arg car_name)/zed/depth/image_rect_color /$(arg car_name)/zed/left/camera_info /$(arg car_name)/zed/left/image_rect_color /$(arg car_name)/zed/right/camera_info /$(arg car_name)/zed/right/image_rect_color /$(arg car_name)/left_right /$(arg car_name)/model_name"/>
    
    <node name="rosbag_record_node" pkg="bair_car" type="rosbag_record_node.py" output="screen"
     args="$(arg car_name) $(arg bagpath) $(arg rosbag_args)" respawn="true" if="$(arg record)">
    </node>
    
</launch>
